docs/design/key_classes.md

```diff
@@ -0,0 +1,55 @@
+# Key classes
+
+## `DetectedObject`
+
+This class represents a detected object received from the host agent (HA) or a remote agent (RA).
+
+### Member variables
+
+* `timestamp`: global time associated with the `state`
+* `state`: state vector (CTRV model)
+* `state_covariance`: covariance matrix associated with `state`
+
+### Possible implementation
+
+```cpp
+struct {
+    namespace cp = cooperative_perception;
+
+    std::chrono::time_point timestamp;
+    cp::CtrvState state;
+    cp::CtrvStateCovariance covariance;
+};
+```
+
+## `DetectedObjectList`
+
+This class contains a list of `DetectedObject`s. We use a custom type instead of directly using a container type (e.g., `std::vector`) because we have size constraints to ensure timing requirements are met.
+
+### Possible implementation
+
+```cpp
+using DetectedObjectList = boost::container::static_vector<DetectedObject, 200>;
+```
+
+## `RemoteAgent`
+
+This class represents a remote agent that shares its own state and detected objects.
+
+### Member variables
+
+* `timestamp`: global time associated with the `state`
+* `state`: state vector (CTRV model)
+* `state_covariance`: covariance matrix associated with `state`
+
+### Possible implementation
+
+```cpp
+struct {
+    namespace cp = cooperative_perception;
+
+    std::chrono::time_point timestamp;
+    cp::CtrvState state;
+    cp::CtrvStateCovariance covariance;
```

Comment:
Remote agent and detected object have an identical content. Maybe there's an intermediate structure which could represent the fields shared by each or an inheritance structure here?

Reply:
I agree there is some shared content, but I think creating an intermediate structure at this point would be premature. We can refactor once there is a larger code base and the common usage is clearer.